addons/sale/models/sale.py

```diff
@@ -397,7 +397,7 @@ def onchange_partner_id(self):
         }
         user_id = partner_user.id
         if not self.env.context.get('not_self_saleperson'):
-            user_id = user_id or self.env.context.get('default_user_id', self.env.uid)
+            user_id = self.env.context.get('default_user_id', user_id or self.env.uid)
```

Comment:
This is a strange change. Indeed default is used when creating and no other value is found. Here we are in an onchange, seems strange to force default from context in an onchange if we have a better option which is the customer's responsible (partner_id.user_id). What is exactly the source of this change ?

Reply:
Hello @tde-banana-odoo ,

It is asked in the spec that when leads are converted to quotation, we should always give priority to the salesperson of lead over the salesperson set on partner. So if you direct create SO, it will work as expected (because we won't have anything in context). Do you have another solution in mind for this?

Thanks in advance!

Reply:
I would not change current behavior which is valid in all use cases. We could imagine supporting a ``sale_force_user_id`` context key and using it like
```
user_id = self.env.context.get('sale_force_user_id') or partner_user.id
if not self.env.context.get('not_self_saleperson'):
    user_id = user_id or self.env.context.get('default_user_id', self.env.uid)
if user_id and self.user_id.id != user_id:
    values['user_id'] = user_id
```